src/server/src/main/java/io/cassandrareaper/service/Heart.java

```diff
@@ -0,0 +1,188 @@
+
+          Set<String> metrics = storage.getNodeMetrics().stream()
+              .map(nodeMetric -> nodeMetric.getHostAddress())
+              .collect(Collectors.toSet());
+
+          forkJoinPool.submit(() -> {
+            context.storage.getClusters().parallelStream().forEach(cluster -> {
+
+              try (Timer.Context t1 = timer(context, "metrics", cluster.getName())) {
+                Set<String> updatedNodes = Sets.newConcurrentHashSet(metrics);
+
+                try (JmxProxy seedProxy = context.jmxConnectionFactory.connectAny(cluster, jmxTimeoutSeconds)) {
+
+                  seedProxy.getLiveNodes().parallelStream()
+                      .filter(node -> 0 <= hostConnectionCounters.getSuccessfulConnections(node))
+                      .filter(node -> updatedNodes.add(node))
+                      .forEach(node -> {
+
+                        try (
+                            Timer.Context t2 = timer(context, "metrics", cluster.getName(), node);
+                            JmxProxy nodeProxy = context.jmxConnectionFactory.connect(node, jmxTimeoutSeconds)) {
+
+                          assert nodeProxy.getHost().equals(node);
+
+                          storage.storeNodeMetrics(
+                              NodeMetrics.builder()
+                                  .withHostAddress(node)
+                                  .withDatacenter(nodeProxy.getDataCenter())
+                                  .withPendingCompactions(nodeProxy.getPendingCompactions())
+                                  .withHasRepairRunning(nodeProxy.isRepairRunning())
+                                  .withActiveAnticompactions(0) // for future use
+                                  .build());
+                        } catch (ReaperException | InterruptedException | JMException ex) {
+                          LOG.warn("failed NodeMetrics update on " + node + " in " + cluster.getName(), ex);
+                        }
+                      });
+
+                  LOG.info(
+                      "Updated metrics in cluster {} for the nodes {}",
+                      cluster.getName(),
+                      String.join(",", updatedNodes));
+                } catch (ReaperException | RuntimeException ex) {
+                  LOG.warn("failed seed connection in cluster " + cluster.getName(), ex);
+                }
+              }
+            });
+          }).get();
+        } catch (ExecutionException | InterruptedException | RuntimeException ex) {
+          LOG.warn("failed updateAllReachableNodeMetrics submission", ex);
+        } finally {
+          assert updatingNodeMetrics.get();
+          updatingNodeMetrics.set(false);
+        }
+      });
+    }
+  }
+
+  private static Timer.Context timer(AppContext context, String... names) {
+    return context.metricRegistry.timer(MetricRegistry.name(Heart.class, names)).time();
+  }
+
+  private void registerGuages(AppContext context) throws IllegalArgumentException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)